protocols/gossipsub/src/metrics.rs

```diff
@@ -49,13 +49,48 @@ pub struct Config {
     /// determined by users on the network.  This limit permits a fixed amount of topics to allow,
     /// in-addition to the mesh topics.
     pub max_never_subscribed_topics: usize,
+    /// Buckets used for the score histograms.
+    pub score_buckets: Vec<f64>,
+}
+
+impl Config {
+    /// Create buckets for the score histograms based on score thresholds.
+    pub fn buckets_using_scoring_thresholds(&mut self, params: &crate::PeerScoreThresholds) {
+        self.score_buckets = vec![
+            params.graylist_threshold,
```

Comment:
I am not sure how valuable values calculated based on the bucket values are if the buckets themselves don't follow some consistent distribution. Would it not make sense to take the smallest of all the values and do an exponential distribution to the max value (`100.0`)?

Reply:
You probably know best from your current measurements. How well does the histogram mirror real world values?

Reply:
The value here is to understand how often nodes are treated in a special way because they have passed the thresholds, more than to understand the distribution of scores itself. In a way is more a measure of network health/score param behaviour than a measure of the scores